### PR TITLE
Update .NET 9 SDK and resolve warnings

### PR DESCRIPTION
**Pull request by sebastienros**
Obsolete method in .NET 9.

---

Fixes https://github.com/grpc/grpc-dotnet/issues/2386

Update:

* Use most recently released .NET 9 SDK
* Fix warnings
* Remove use of unlisted testing package
* Fix build because of docker-compose v1 removal - https://github.com/actions/runner-images/issues/9692

### CODE REVIEW

**Comment by sebastienros** [comment]
@JamesNK

File: .gitignore [hunk 1]
```diff
@@ -34,7 +34,6 @@ launchSettings.json
 BenchmarkDotNet.Artifacts/
 BDN.Generated/
 binaries/
-global.json
```
Comment (sebastienros) [inline .gitignore:37]: Not sure what was the reason.

File: global.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "sdk": {
-    "version": "9.0.100-preview.5.24305.3",
+    "version": "9.0.100-rc.1.24413.1",
```
Comment (sebastienros) [inline global.json:3]: Very recent change, so trying with the nightly build.

**Comment by JamesNK** [comment]
Added some updates required to make the build pass. Reaction to https://github.com/actions/runner-images/issues/9692

**Review by JamesNK — APPROVED** [review]

**Pull request merged**